client/state/user-settings/thunks/save-two-step-sms-settings.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import wp from 'calypso/lib/wp';
+
+/**
+ * Internal dependencies
+ */
+import {
+	saveUserSettingsFailure,
+	saveUserSettingsSuccess,
+} from 'calypso/state/user-settings/actions';
+import { fromApi } from 'calypso/state/data-layer/wpcom/me/settings';
+import { USER_SETTINGS_SAVE } from 'calypso/state/action-types';
+
+const wpcom = wp.undocumented();
+
+const saveTwoStepSMSSettings = ( countryCode, phoneNumber ) => async ( dispatch ) => {
```

Comment:
I purposely did not use the data layer here. The reason for this is that the two step authentication setup flow currently is a mix of internal component state and redux and relies on a callback after the two step phone number got saved. We can change that, however, that'd be a more complex refactor which I think is out of scope for a general reduxification of `lib/user-settings` – that said I'd like to follow up on this with another PR.

Reply:
Totally agree with the sentiment here and I also prefer this implementation, given the circumstances.